### PR TITLE
fix(clients): post-#873 scope follow-ups — test hermeticity, sweep transparency, probe coverage

### LINKED ISSUE #876: tests: setting godot_ai/mcp_client_scope to project fails the GDScript suite
### Summary

Setting the new `godot_ai/mcp_client_scope` EditorSetting (#872 / #873) to `project` —
its headline value — makes the GDScript suite fail. The failure is in
`test_client_attach_config.gd`, a suite that has nothing to do with the scope feature
and never opts into it.

A developer who actually adopts the feature gets a red suite with no obvious connection
to the setting they changed.

### Reproduction

Verified on Godot `4.7.2.stable` (`ed1daf0bf`), headless editor, plugin `3.1.5`,
at `143e3ba` (current `main`):

| `godot_ai/mcp_client_scope` | full suite |
|---|---|
| `user` (default) | 2161 passed / **0 failed** / 36 skipped |
| `project` | 2160 passed / **1 failed** / 36 skipped |

```
client_attach_config.test_claude_cli_json_fallback_migrates_legacy_http_entry
  Expected 2, got 0
```

(`2` = `CONFIGURED_MISMATCH`, `0` = `NOT_CONFIGURED`.)

### Root cause

`test_client_attach_config.gd:813` `_claude_cli_clone()` builds its fixture by copying
fields off the **real** `claude_code` descriptor:

```gdscript
client.cli_register_template = registered.cli_register_template
client.cli_unregister_template = registered.cli_unregister_template
client.cli_status_args = registered.cli_status_args
```

Since #873 `cli_register_template` carries `{scope}`, so the clone now reports
`CliStrategy.uses_scope_token() == true`. It does **not** copy the new
`cli_scope_status_template`.

At `project` scope, in `_dispatch_check_status_with_cli_path_details`:

1. `_scope_diverges_from_json_fallback` (`client_configurator.gd:762`) returns `true`
   — token present, scope is not the default, and the test passes a non-empty
   `cli_path` (`"/fake/claude-not-invoked"`).
2. So the JSON-fallback read at `client_configurator.gd:672-677` is skipped — which is
   the whole point of that test ("proving the JSON file is authoritative for
   command-shape CLI clients").
3. The scope-probe gate at `client_configurator.gd:694-700` requires a non-empty
   `cli_scope_status_template`, which the clone lacks, so it does not fire either.
4. Control reaches `CliStrategy.check_status_details` with a binary that does not
   exist → `spawn_failed` → `NOT_CONFIGURED`.

`test_client_attach_config.gd` contains zero references to `SETTING_CLIENT_SCOPE`, so it
neither pins nor restores the setting. `test_clients.gd` has `_restore_client_scope()`,
but it restores the developer's *saved* value rather than the default — so an adopted
`project` setting persists across the run.

### Suggested fix

Make the suite hermetic rather than dependent on ambient editor state: pin
`godot_ai/mcp_client_scope` to `McpSettings.DEFAULT_CLIENT_SCOPE` for the duration of
`test_client_attach_config.gd` and restore the previous value in teardown.

More generally, any suite that clones a real descriptor is now exposed to this setting;
a shared save/pin/restore helper would stop the next one silently inheriting it.

### Environment

- godot-ai `3.1.5` @ `143e3ba`
- Godot `4.7.2.stable (official)`
- Windows 11

### LINKED ISSUE #877: clients: Configure's all-scope sweep silently rewrites .mcp.json in the editor's cwd, and the dock never shows the removes
### Summary

Configure's all-scope pre-cleanup (#873) runs `claude mcp remove --scope project godot-ai`
on **every** Configure, including at the default `user` scope. That command deletes the
`godot-ai` key from the `.mcp.json` in the spawned CLI's working directory — which is
whatever directory the editor process happened to inherit at launch, not necessarily the
Godot project.

The sweep itself is deliberate and documented (`README.md:122`: "clears `godot-ai` out of
every scope before writing the new one"). What is missing is any connection between that
line and the uncontrollable-cwd caveat documented separately, plus any surfacing of the
removes in the UI.

### Reproduction

Real `claude` 2.1.241, isolated `CLAUDE_CONFIG_DIR`, a throwaway git repo with a
committed `.mcp.json`:

```json
{
  "mcpServers": {
    "godot-ai": { "type": "stdio", "command": "uvx",
                  "args": ["godot-ai", "attach", "--hand-tuned-by-the-team"] },
    "some-other-server": { "type": "stdio", "command": "node", "args": ["other.js"] }
  }
}
```

```
$ claude mcp remove --scope project godot-ai
Removed MCP server godot-ai from project config
File modified: .../fakerepo/.mcp.json
exit 0
```

Result: the hand-tuned `godot-ai` entry is gone, the file is dirty in git, and
`some-other-server` was reformatted as collateral. Exit code is 0, and
`_cli_strategy.gd:57` discards the result, so the plugin cannot notice.

### Where

`plugin/addons/godot_ai/clients/_cli_strategy.gd:45-57` — the loop is guarded only on a
non-empty unregister template. There is no scope-equality check:

```gdscript
for pre_scope in _cleanup_scopes(client):
    var pre_args := _format_args(
        client.cli_unregister_template, server_name, server_url, {}, pre_scope
    )
    McpCliExec.run(cli, pre_args, _REMOVE_TIMEOUT_MS)
```

`_cleanup_scopes` returns all of `CLIENT_SCOPES` for any `{scope}` descriptor, pinned by
`test_clients.gd` `test_client_scope_pre_cleanup_sweeps_every_scope`. `claude_code.gd:23-30`
concedes the cwd problem: *"`--scope project` resolves `.mcp.json` against the spawned
CLI's working directory ... Godot has no API to set a child's cwd."*

### Scope of the problem (what it is *not*)

- Only the `project` pass can touch a version-controlled file. `--scope local` writes a
  per-project block inside `~/.claude.json`; it does not touch `.mcp.json`.
- Blast radius is one key named exactly `godot-ai`, not the whole file.
- `mcp remove` on an absent entry really is a harmless no-op, as the comment claims.

So this is not "Configure corrupts your repo". It is: at the default setting, Configure
has a silent, uncommunicated write to a file in an unrelated directory that a team may
be tracking in git.

### Suggested fix

Transparency rather than behaviour change:

1. **Show the removes in the dock's "Run this manually" text.**
   `clients/_manual_command.gd` `_build_cli` renders only `cli_register_template`, so the
   snippet a user is told to run is not what Configure runs. Rendering the sweep lines
   makes the side effect self-documenting and keeps the manual path faithful.
2. **Connect the two README bullets** so the sweep and the cwd caveat are read together.
3. Optionally, skip the `project` pass when the selected scope is `user` and no
   `{scope}` entry was ever written at project scope — though this trades the
   stranded-entry guarantee the sweep exists to provide, so (1) and (2) may be enough.

### Environment

- godot-ai `3.1.5` @ `143e3ba`
- Claude Code CLI `2.1.241`, Windows 11

### LINKED ISSUE #878: tests: scope-aware status probe never executes in the suite; two mutations survive
### Summary

The scope-aware status probe added in #873 — `CliStrategy.check_scope_status_details`
(`_cli_strategy.gd:247-275`) and its dispatch gate (`client_configurator.gd:694-700`) —
never executes during the test suite. Only the pure helpers `_scope_probe_verdict` and
`_scope_from_probe_output` are covered.

### Evidence

Instrumented a full headless run on Godot `4.7.2.stable`, plugin `3.1.5` @ `143e3ba`,
and counted the synthetic client's subprocess spawns in the editor log:

```
godot-ai-nonexistent-cli-xyz mcp list   3
godot-ai-nonexistent-cli-xyz mcp get    0
```

`mcp list` is `cli_status_args`; `mcp get` is `cli_scope_status_template`. Zero `mcp get`
spawns means the scope-probe branch was never taken.

Cause: `test_clients.gd:1531` `_make_scope_cli_client()` sets `cli_register_template`,
`cli_unregister_template` and `cli_status_args`, but never `cli_scope_status_template`.
The gate at `client_configurator.gd:694-700` requires that field to be non-empty, so
control always falls through to the plain `check_status_details`.

### Mutations that survive the whole suite

Both silently disable scope detection for the only client that has the feature:

1. `client_configurator.gd:699` — `McpSettings.client_scope()` → `McpSettings.DEFAULT_CLIENT_SCOPE`
   (the probe would always compare against `user`).
2. `_cli_strategy.gd:264` — `client.cli_scope_status_template` → `client.cli_status_args`
   (the probe would run `mcp list`, whose output has no `Scope:` line, and every verdict
   would degrade to the target check).

### Fairness note

Most of `check_scope_status_details` is a structural clone of the already-covered
`check_status_details` — same `cli.is_empty()` guard, same `command_launch_error`
fail-close, same `timed_out` / `spawn_failed` branches. The real decisions live in
`_scope_probe_verdict`, which *is* tested against verbatim `claude mcp get` output. So
this is duplicated plumbing left uncovered rather than a body of untested logic — but the
two mutations above are genuine holes, and the gate itself is what routes them.

### Suggested fix

- Set `cli_scope_status_template` on `_make_scope_cli_client` so the dispatch gate is
  actually reachable from a test, and assert the routed spawn-failure path.
- Add a case that would fail under mutation (1) — e.g. assert the probe is invoked with
  the *selected* scope rather than the default.
- Consider asserting on the rendered argv (`format_args(cli_scope_status_template, ...)`)
  so mutation (2) is caught without needing a live binary.

### Related

Coverage gaps in the same area, for whoever picks this up: `client_scope()`'s
`value == null` branch (`settings.gd:99`), `_client_scope_setting()`'s worker-thread
branch, `warm_client_scope()`, `format_args`'s `scope_override` parameter, the
`configure` all-scope cleanup loop, and `_note_unhonoured_scope`'s token guard all have
no direct test.

### LINKED ISSUE #879: cleanup after #873: dead PATH-walk clause, wrong path hint on cross-scope mismatch, stale descriptor comments
Three small follow-ups from reviewing #873 after merge. None change behaviour except (2).

### 1. `_scope_diverges_from_json_fallback`'s PATH walk is dead code, and its docstring says otherwise

`client_configurator.gd:767-768`:

```gdscript
var resolved := cli_path if not cli_path.is_empty() else CliStrategy.resolve_cli_path(client)
return not resolved.is_empty()
```

The docstring above it (`:757-761`) says *"the PATH walk only runs when the answer can
actually be true"*, implying the clause is load-bearing. It cannot change any outcome:

- The function is only consulted from `client_configurator.gd:674`, inside a condition
  that already requires `client.has_json_fallback()`.
- If the walk returns empty, the function returns `false` and the JSON read happens at
  `:672-677`.
- If it is forced to `true` instead, the JSON read is skipped, but `:682`
  (`if resolved_cli.is_empty() and client.has_json_fallback():`) takes the fallback
  anyway — same status, same file.

So the clause costs a second full `McpCliFinder.find` PATH sweep per status check (the
same walk runs again at `:679`) and buys nothing. Either drop it and return `true`, or
thread the already-resolved value down instead of re-walking. Either way the docstring
needs to stop claiming it is decisive.

### 2. Verification error names the wrong config file after a cross-scope mismatch

`client_configurator.gd:809`:

```gdscript
var path := client.resolved_config_path()
var path_hint := "" if path.is_empty() else " Inspect %s and remove the godot-ai entry by hand if needed." % path
```

`resolved_config_path()` is `path_template`, i.e. `~/.claude.json`. When the surviving
entry is the *project*-scope one, the message tells the user to inspect `~/.claude.json`
where there is nothing to find — the entry is in `<cwd>/.mcp.json`.

The probe already knows which scope actually resolved (`_scope_probe_verdict` puts it in
`error_msg`, e.g. `"registered at user scope, not project"`), so the hint can name the
right location instead of assuming the fallback file.

### 3. Stale descriptor comments in `clients/claude_code.gd`

These describe pre-#873 behaviour and now contradict the code they sit on:

- `:20-21` — *"Explicit scope: an unscoped `mcp remove` deletes from whichever scope
  matches first, which could eat a project-local entry the user made"* — sits directly
  above `cli_unregister_template`, which the Configure sweep now runs at **every** scope,
  including that project-local one.
- `:46` and `:56-57` — still spell the command as `claude mcp add --scope user`.
- `:51-53` — *"Status always reads this file — it is the CLI's own store for user scope"*
  — false at project/local scope, where `_scope_diverges_from_json_fallback` routes status
  to the CLI probe instead.

These are the primary documentation for the descriptor and were left untouched while
every line around them changed.

### PR DESCRIPTION
**Pull request by azereki**
Follow-ups from reviewing #873 after it merged.

Closes #876, closes #877, closes #878, closes #879.

> **Credit:** @dsarno's #881 fixed the same four issues independently and was closed unmerged. Three of its approaches were better than mine and are adopted here — the fake-CLI probe test, the structured `resolved_scope` hint, and the single-shell-block manual command. Each is marked below and in the relevant commit.

### #876 — suite red at `mcp_client_scope = project`

`_claude_cli_clone` builds its fixture from the real `claude_code` descriptor, so since #872 it inherits `{scope}` and reports `uses_scope_token() == true` — but it does not copy `cli_scope_status_template`. At `project`/`local` that flips `_scope_diverges_from_json_fallback`, skipping the JSON read the suite exists to prove is authoritative, while the scope probe does not fire either, so the fake CLI just spawn-fails.

| `godot_ai/mcp_client_scope` | full suite (before) |
|---|---|
| `user` | 2161 passed / **0 failed** |
| `project` | 2160 passed / **1 failed** |

The suite now pins the default for its run and restores the developer's value in teardown. Presence is tracked separately from value in both `test_client_attach_config.gd` and `test_clients.gd` (scope **and** both ports), so a key that was absent is erased rather than created — the repo's existing idiom from `test_allow_hosts.gd` and `test_dock.gd`.

### #877 — the all-scope sweep was invisible

Configure runs the unregister template once per scope before registering, so pressing it at **any** setting — the default `user` included — runs `claude mcp remove --scope project godot-ai`. That rewrites the `.mcp.json` in the CLI's working directory, which is wherever the editor was launched from. Confirmed against claude 2.1.241: a hand-maintained `godot-ai` entry in a checked-in `.mcp.json` is deleted, the file is left dirty, and the command exits 0 so the discarded result cannot tell us anything happened.

The sweep is intended and stays unchanged. What changed is that it is now visible: the dock's "Run this manually" text renders the removes and the register line under **one** shell label in the order Configure runs them (structure from #881), with the cwd hazard as a caveat outside the block so pasting never picks up prose. README states the consequence where it describes the sweep.

### #878 — the scope probe never ran in the suite

`_make_scope_cli_client` never set `cli_scope_status_template`, so the dispatch gate was unreachable and `check_scope_status_details` never executed — a full run spawned `mcp list` three times and `mcp get` zero times.

**Correction to an earlier version of this PR:** the first test I added here did *not* kill the two mutations its commit message claimed. It asserted on templates rendered directly via `format_args` and drove the dispatch only with an unspawnable binary, which proves reachability and fail-closed behaviour and nothing else. That claim was asserted, not tested.

It is now replaced (technique from #881) with a fake CLI that records its argv and prints the recorded `claude mcp get` fixtures, so the real subprocess path runs. Verified by actually applying each mutation on Godot 4.7.2:

| mutation | result |
|---|---|
| `client_configurator.gd:699` `client_scope()` → `DEFAULT_CLIENT_SCOPE` | **FAIL** — "a user-scope entry under a project selection must read as mismatch" |
| `_cli_strategy.gd:264` `cli_scope_status_template` → `cli_status_args` | **FAIL** — "'get' not found in 'mcp list'" |

Both reverted; suite green with them restored.

### #879 — cleanup

- `_scope_diverges_from_json_fallback`'s PATH walk could not change any outcome — one call site, already guarded on `has_json_fallback()`, and the next branch takes the fallback read when the CLI does not resolve. It only cost a second `McpCliFinder.find` sweep per status check. Dropped, with the unused parameter and the docstring that claimed it was decisive.
- `_verify_post_state` pointed at `resolved_config_path()` (`~/.claude.json`), the wrong file when the survivor is project-scoped. The probe now carries `resolved_scope` as structured data and `_post_state_path_hint` picks the matching location (approach from #881, replacing the prose hint I had).
- Three `claude_code.gd` comment blocks still described pre-#873 behaviour.

### Verification

Full gauntlet on **Godot 4.7.2-stable** (`ed1daf0bf`), headless editor, isolated self-contained config and non-default ports:

- `ruff check src/ tests/` — clean
- `pytest -q` — 1766 passed, 19 skipped, 0 failed
- `script/ci-check-gdscript` — all GDScript files OK
- `test_run` — **2164 passed / 0 failed / 36 skipped across 69 suites**, run at both `user` and `project` scope

One environment note for reviewers: on a freshly created worktree the first editor start can register only part of the test tree (I saw 68 suites / 2171 tests, with `clients` reporting 163 of 174, no load errors and nothing failing). Re-importing and restarting the editor restores the full 69. That is the silent-shortfall case filed separately as #882; it is not caused by this branch, but it does mean a single short run can under-report.

## Summary by CodeRabbit

- **New Features**
  - Manual configuration commands are now provided in one copy-pasteable block, including cleanup commands for all applicable scopes.
  - Successful configuration displays which scopes were cleared in the client status row.
  - Status checks and post-action guidance identify project, local, and user scopes more accurately.

- **Bug Fixes**
  - Improved detection of scope mismatches and configured entries, reducing misleading status results and cleanup instructions.

- **Documentation**
  - Added an important notice explaining that Configure removes `godot-ai` entries across client scopes and may modify project configuration files.

### CODE REVIEW

File: test_project/tests/test_client_attach_config.gd [hunk 1]
```diff
@@ -4,6 +4,7 @@ extends McpTestSuite
 ## #816 rollout steps 5-6: launch discovery plus Codex's TOML command shape.
 
 var _scratch_dir: String
+var _saved_client_scope: Variant = null
```
[the remark this answers is not in the document]
Comment (azereki) [reply test_project/tests/test_client_attach_config.gd:8]: Fixed in 0bd56bf — presence is now tracked separately from value, and teardown erases the key when it was absent at setup.

Worth recording why, since the reachability and the fix point in opposite directions:

The absent branch cannot be hit today. `plugin.gd._enter_tree` calls `ClientConfigurator.ensure_settings_registered()`, which seeds `godot_ai/mcp_client_scope` via `_register_client_scope_setting`, and the GDScript suite only runs inside a live editor with the plugin enabled. So the key always exists by the time `suite_setup` runs.

I fixed it regardless, because the repo already has the presence-aware idiom in two places — `test_allow_hosts.gd` (`_had_allow_hosts_setting` + `es.erase`) and `test_dock.gd::_restore_mcp_logging_setting` — and what I wrote was copied from the weaker `test_clients.gd::_restore_client_scope`, which unconditionally writes the default. Following the stronger of two in-repo conventions is the right call, and a restore helper that only round-trips correctly in the state it happens to meet is the kind that breaks silently when that state changes.

I did **not** add the suggested coverage for both branches. That would mean asserting on the test harness's own teardown, and the absent case can't be constructed from inside a running editor without unregistering the plugin's settings first — the test would have to fake the thing it was checking.

Note `test_clients.gd::_restore_client_scope` has the same unconditional-write shape. I left it alone here to keep this PR's diff scoped, but it's the same latent issue if anyone wants it as a follow-up.

Verified on Godot 4.7.2 headless: 2162 passed / 0 failed / 36 skipped across 69 suites; `ruff` clean; `pytest` 1766 passed / 19 skipped.

**Comment by azereki** [comment]
Going to take some #881 fixes that are materially stronger than mine before calling this good

**Review by dsarno — COMMENTED** [review]
## Review (correctness + verification)

Verified the whole diff at `6549d1f`. This is in excellent shape — the four issue fixes are all correct, and the pieces adopted from #881 landed cleanly (thanks for the careful crediting, and for the honest correction on the earlier #878 test).

**Independently verified:**

- The target-drift gap CodeRabbit flagged on #881 (and I carried over here) is genuinely fixed: `_scope_probe_verdict` now parses `Scope:` before the target check and attaches `resolved_scope` on **both** mismatch paths, with test legs for drift-with-label and drift-without-label. That was the one behavioral thing I wanted to confirm before this merged.
- `_post_state_path_hint` matches probe reality per scope, and the empty-`resolved_config_path` guard on the `local` branch is a case #881's version missed.
- Ran the full GDScript suite against this head on Linux with ambient `godot_ai/mcp_client_scope = project`: **2175/2202 passed, 0 failed** — the #876 fix confirmed on a second platform (complementing your Windows runs). `ci-check-gdscript` clean.
- The dock sweep disclosure is a good call — the manual panel only shows on failure, so the success path (where the sweep actually ran) was indeed silent.

**Two one-liners still open from CodeRabbit's last pass** (both valid, both trivial):

1. `configure_sweep_note` says "cleared" but `McpCliStrategy.configure` discards every pre-cleanup result — a timed-out or failed remove leaves the scope untouched while the row claims it was cleared. "attempted to clear" (or similar) keeps the note honest.
2. `test_dock.gd`'s new test has a bare `return` when the `claude_code` row is missing — per the repo's test guideline, that path needs an assertion or `skip("reason")` so the regression test can't silently pass with zero assertions.

With those two folded in, this is ready to merge from my side.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: plugin/addons/godot_ai/client_configurator.gd [hunk 2]
```diff
@@ -810,6 +812,55 @@ static func _verify_post_state(
...
+## disclosure. Empty for descriptors without a scope token: their single
+## implicit pass removes exactly the entry the register is about to rewrite,
+## which needs no warning (the same rule `_sweep_caveat` applies).
+static func configure_sweep_note(id: String) -> String:
+	var client := ClientRegistry.get_by_id(id)
+	if client == null or client.config_type != "cli":
+		return ""
+	if client.cli_unregister_template.is_empty() or not CliStrategy.uses_scope_token(client):
+		return ""
+	return "cleared %s from %s" % [SERVER_NAME, ", ".join(CliStrategy.cleanup_scopes(client))]
```
[the remark this answers is not in the document]
Comment (azereki) [reply plugin/addons/godot_ai/client_configurator.gd]: Fixed in 0c481d9 — the note now reads `attempted to clear godot-ai from user, project, local`.

Checking the discarded remove results would not have been enough on its own: `mcp remove` exits non-zero for an absent entry as well as for a real failure, so exit status cannot separate "nothing was there" from "the remove failed". The wording is the honest option. README says the same, the dock test now asserts the row carries the note itself rather than the literal word, and the clients test pins the best-effort phrasing so it cannot drift back.

File: test_project/tests/test_dock.gd [hunk 3]
```diff
@@ -593,6 +593,39 @@ func test_drift_banner_clears_after_per_row_reconfigure() -> void:
...
 
+func test_successful_configure_discloses_the_scope_sweep_on_the_row() -> void:
+	## #877: `_show_manual_command_for` — the only thing that reveals the panel
+	## listing the pre-cleanup removes — is called just once, on the Configure
+	## FAILURE branch. So a successful Configure cleared `godot-ai` out of every
+	## scope, including a .mcp.json in whatever directory the editor happened to
+	## be launched from, and told the user nothing. The green row carries that.
+	_dock._build_ui()
+	if not _dock._client_rows.has("claude_code"):
+		return
```
[the remark this answers is not in the document]
Comment (azereki) [reply test_project/tests/test_dock.gd:606]: Fixed in 0c481d9. Went with `assert_true` rather than `skip()`: `claude_code` is a static registry entry (`clients/_registry.gd`), so a missing row is a real failure, not an environment precondition — the same treatment `test_clients.gd` already gives it with `assert_true(client != null, "claude_code must be registered")`.

**Comment by azereki** [comment]
Both remaining items folded in at 0c481d9 (details in the two thread replies): the sweep note is now "attempted to clear …" with README/tests aligned, and the dock regression test asserts the `claude_code` row before it can return.

Re-verified on Godot 4.7.2-stable, headless editor: `ruff` clean, `pytest` 1766 passed / 19 skipped, `ci-check-gdscript` OK, `test_run` 2168/2202 passed, 0 failed, 34 skipped. The two edited tests were also re-run by name — 1/1 each, 0 skipped — so neither is passing on the #882 shortfall.

**Review by dsarno — APPROVED** [review]
Verified `0c481d9`: both items fixed exactly as discussed, each with a test pin so the wording and the row assertion can't drift back — and your reasoning on both is right (`mcp remove`'s exit code genuinely can't separate "absent" from "failed", so honest wording beats result-checking; and `assert_true` over `skip()` is correct for a static registry entry). The README now states the best-effort nature in the same breath as the disclosure, which is the most honest version of this whole feature.

Approving. Great work on this one — the mutation-verified test claims and the cross-pollination with #881 made it a pleasure to review.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by dsarno** [comment]
Merged — thank you @azereki! Four issues filed and fixed within a day of #873 landing, with real-CLI repros, mutation-verified tests, and gracious handling of the #881 collision. Exactly the kind of contribution that makes this project better.

---
_Generated by [Claude Code](https://claude.ai/code)_
